payjoin/src/persist/mod.rs

```diff
@@ -0,0 +1,74 @@
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::uri::ShortId;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum PersistKey {
+    Url(Url),
+    ShortId(ShortId),
+}
```

Comment:
I'm confused about the coupling here. Why does Persist need to know about the type of keys it's dealing with? Shouldn't that be parametrized by the caller?

Reply:
Persist the mod doesnt really need knowledge of the keys its dealing with. That would kinda defeat the purpose of the associated type (Key). This is used by the noopPersister only -- which happens to live in this mod.